engines/engine.cpp

```diff
@@ -452,41 +452,50 @@ int initGraphicsAny(const Graphics::ModeWithFormatList &modes, int start) {
 	return candidate;
 }
 
-void initGraphics(int width, int height, const Graphics::PixelFormat *format) {
+void initGraphics(int width, int height, const Graphics::PixelFormat *format, bool required) {
 	Graphics::ModeWithFormatList modes;
-	modes.push_back(Graphics::ModeWithFormat(width, height, format));
+	if (!format) {
+		// Use any pixel format
+		modes.push_back(Graphics::ModeWithFormat(width, height, nullptr));
+	} else if (required) {
+		// Try the provided format, and fail if it's not available
+		modes.push_back(Graphics::ModeWithFormat(width, height, format));
+	} else {
+		// Prefer the provided format if it's supported by the backend
+		const Common::List<Graphics::PixelFormat> backendList = g_system->getSupportedFormats();
+		if (Common::find(backendList.begin(), backendList.end(), *format) != backendList.end()) {
+			modes.push_back(Graphics::ModeWithFormat(width, height, format));
+		}
+		// Otherwise use any pixel format
+		modes.push_back(Graphics::ModeWithFormat(width, height, nullptr));
+	}
 	initGraphicsAny(modes);
 }
 
-/**
- * Determines the first matching format between two lists.
- *
- * @param backend	The higher priority list, meant to be a list of formats supported by the backend
- * @param frontend	The lower priority list, meant to be a list of formats supported by the engine
- * @return			The first item on the backend list that also occurs on the frontend list
- *					or PixelFormat::createFormatCLUT8() if no matching formats were found.
- */
-inline Graphics::PixelFormat findCompatibleFormat(const Common::List<Graphics::PixelFormat> &backend, const Common::List<Graphics::PixelFormat> &frontend) {
-#ifdef USE_RGB_COLOR
-	for (const auto &back : backend) {
-		for (auto &front : frontend) {
-			if (back == front)
-				return back;
+void initGraphics(int width, int height, const Common::List<Graphics::PixelFormat> &formatList, bool required) {
+	Graphics::ModeWithFormatList modes;
+	if (required) {
+		// Try all provided formats, and fail if none are available
+		for (const auto &format : formatList) {
+			modes.push_back(Graphics::ModeWithFormat(width, height, format));
 		}
+	} else {
+		// Prefer any of the provided formats if it's supported by the backend
+		const Common::List<Graphics::PixelFormat> backendList = g_system->getSupportedFormats();
```

Comment:
Here, I am not sure.
You select the preferred engine format while we could select the preferred backend format.
I suppose that if an engine provides several formats, it means that it doesn't do any software conversion with them, it's only a matter of which pixel format to use when decompressing data.
On the other hand, the preferred pixel format provided by the backend may still be a better match regarding the device capabilities (for example, OpenGLGraphicsManager could report at first the defaultFormatAlpha/defaultFormat). Then engine would then select it and this may avoid conversions in the OpenGL stack.